src/python/foglamp/data_purge/purge.py

```diff
@@ -107,57 +107,14 @@
 """
 
 
-def get_nth_id() -> int:
-    """Update the config file to have row ID somewhere within the oldest 100 rows.
-    This method would potentially be replaced by a mechanism that is aware what was the last value sent to the
-     historian.
-    Returns:
-        Method doesn't return anything
+def get_last_id() -> int:
     """
-    rand = random.randint(1, 100)
-    stmt = "SELECT id FROM (SELECT id FROM readings ORDER BY id ASC LIMIT %s)t ORDER BY id DESC LIMIT 1"
-    row_id = execute_command(stmt % rand).fetchall()
-    try:
-        return int(row_id[0][0])
-    except IndexError:
-        return 1
-
-
-def convert_timestamp(set_time: str) -> datetime.timedelta:
-    """Convert "age" in config file to timedelta. If only an integer is specified,  then 
-        the code assumes that it is already in minutes (ie age:1 means wait 1 minute) 
-    Args:
-        set_time (str): Newest amount of  time back to delete
-    Returns:
-        converted set_time to datetime.timedelta value
+    From foglamp.streams retrive the heighest last_object that corresponds to the newest readings table row id sent to Pi/OMF 
+    Returns: 
+        The newest `readings` table row id that was sent to Pi/OMF
     """
-    if type(set_time) is int or set_time.isdigit():
-        return datetime.timedelta(minutes=int(set_time))
-    time_dict = {}
-    tmp = 0
-
-    for value in set_time.split(" "):
-        if value.isdigit() is True:
-            tmp = int(value)
-        else:
-            time_dict[value] = tmp
-
-    time_in_sec = datetime.timedelta(seconds=0)
-    time_in_min = datetime.timedelta(minutes=0)
-    time_in_hr = datetime.timedelta(hours=0)
-    time_in_day = datetime.timedelta(days=0)
-
-    for key in time_dict.keys():
-        if 'sec' in key:
-            time_in_sec = datetime.timedelta(seconds=time_dict[key])
-        elif 'min' in key:
-            time_in_min = datetime.timedelta(minutes=time_dict[key])
-        elif ('hr' in key) or ('hour' in key):
-            time_in_hr = datetime.timedelta(hours=time_dict[key])
-        elif ('day' in key) or ('dy' in key):
-            time_in_day = datetime.timedelta(days=time_dict[key])
-    return time_in_sec+time_in_min+time_in_hr+time_in_day
-
+   stmt = "SELECT MAX(last_object) FROM foglamp.streams;" 
```

Comment:
Using MAX(last_object) FROM fog lamp.steams will mean that we get the latest not sent to any channel. Whereas if we use MIN(last_object) then we get the latest not sent to all channels. There is a subtle but important difference.

However I do not understand why we need a Python function at all. Se can simply do it with the delete statement. If the flag to **not** delete unsent data is true then we add an extra where clause to the delete statement...

and id < (select MIN(last_object) FROM fog lamp.steams) 

So the statement is something like...

deleteunsent = ''
if ! deleteRowsNotSent: 
    deleteuenset = ' AND id < (SELECT MIN(last_object) FROM foglamp.steams)'
stmt = 'DELETE FROM foglamp.readings where DATE_PART(\'day\', now() - user_ts) * 24 + 
              DATE_PART(\'hour\',now() - user_ts) < {0} {1}'.format(age, deleteunsent)


The above Python may not be 100% correct, I'm making it up on the spur of the moment. However the idea is this should be really simple and just come down to creating a SQL statement that does all the work in the database. If Postgres had datediff it would be even easier.